fusedrug/utils/cluster/cluster_using_mmseqs.py

```diff
@@ -0,0 +1,156 @@
+            if force_rebuild:
+                print(f'cluster::Hash value on disk matches the calculated one ({hash_value}), but force_rebuild was requested so rebuilding anyway.')
+                _rebuild_needed = True
+        else:
+            raise Exception('hash value on disk is a mismatch! please delete the directory manually and rerun.')
+            _rebuild_needed = True
+
+        existing_desc = read_text_file(already_created_description_filename)
+    else:
+        _rebuild_needed = True
+            
+    if _rebuild_needed and (not _rebuild_needed):
+        print(f'Hash value indicates that it was already built, not rebuilding. See description: {existing_desc}')
+        return
+
+    cluter_impl(**kwargs)
+
+    save_text_file_safe(already_created_description_filename, hash_value)
+    save_text_file_safe(already_created_hash_filename, str_repr)
+
+def cluter_impl(*, input_fasta_filename:str, output_dir:str, cluster_min_seqeunce_identity:float=0.4, cluster_method:str='cluster'):
+    """
+
+    Uses mmseqs to:
+
+    1. Remove 100% sequence identity duplicates
+    2. Cluster the remaining unique sequences into multiple clusters (e.g. with 70% sequence similary threshold )
+        This is useful for multiple purposes:
+        a. Creating cross validation/test splits that evaluate and demonstrate generalizability
+        b. Balanced sampling during training, sampling with inverse proportion to cluster size. (Similar to Class Balancing)        
+
+    Args:
+        input_fasta_filename: a fasta with an entry per molecular entity
+        output_dir: where the output will be generated
+        cluster_min_seqeunce_identity: the minimal sequence identity for member within a cluster
+        cluster_method: any of 'cluster', 'linclust':
+            'cluster' is the "vanilla" one
+            'linclust' is faster (claims linear runtime) but less accurate. Might be suitable for massive data.
+                NOTE: I've compared cluster and linclust results for the deduplication phase, and results aren't identical, which means it probably misses few identical cases.
+
+        
+    """    
+
+    which_mmseqs = shutil.which('mmseqs')
+
+    if which_mmseqs is None:
+        raise Exception('Please install mmseqs2 . See install instructions here: https://github.com/soedinglab/MMseqs2 '
+        'Also note that you can download prebuilt static binaries such as: https://mmseqs.com/latest/mmseqs-linux-sse41.tar.gz - extract and add the binary to your system PATH.')
+    else:
+        print(f'identified mmseqs binary at: {which_mmseqs}')
+    
+    if not os.path.isfile(input_fasta_filename):
+        if os.path.isfile(input_fasta_filename+'.gz'):
+            cmd = f'gunzip {input_fasta_filename}.gz'
+            _run_system_cmd(cmd)            
+    
+    mmseqs_db_path = os.path.join(output_dir, 'mmseqs_DB')
+
+    print('cluster_method=', cluster_method)
+
+    ########### Major step A - remove all redundancies
```

Comment:
Can we make step A optional (additional boolean flag)? because in my use case, I don't want to remove duplications.